### PR TITLE
Show encryption notification only once

### PR DESCRIPTION
**Pull request by bermeitinger-b**
Show the encryption warning only once.

### CODE REVIEW

**Comment by bb010g** [comment]
Script is still inconsistent between `gnome_libsecret` and `gnome-libsecret`.

**Comment by bermeitinger-b** [comment]
Always showing the warning when basic is selected seems out of scope for us. This should be an upstream discussion.

**Comment by minosimo** [comment]
It is not necessary to warn all users that we are using the unencrypted key store when the application has already been doing so for a long time.

The warning should appear:
- When launching Signal for the first time after a new install
- On an existing install, if the user switches from a system keystore to the basic keystore

**Comment by bbhtt** [comment]
> It is not necessary to warn all users that we are using the unencrypted key store when the application has already been doing so for a long time.

The difference is previously the native package didn't support it and official builds uses plaintext as well. Now the official ones don't do anymore

This makes Flatpak a worse option for some people. Users should be aware of that, new or old both and choose what's best for them.

**Comment by bermeitinger-b** [comment]
> On an existing install, if the user switches from a system keystore to the basic keystore
This will never be intentional, right? Why would they change a running system and intentially lessen the security. It could happen, if they somehow reset the env var to default. But then, the warning is useless because Signal will complain anyway.

Is there a nice way to check if this is the first install? Then do this instead of the warning_shown file.


> The difference is previously the native package didn't support it and official builds used plaintext as well. Now the official ones don't do anymore
They fallback to basic just as we do here. Apparently, the implementation is a bit broken, so it can't find the correct backend correctly. This database error is not exclusive to flatpak.

**Comment by minosimo** [comment]
> > On an existing install, if the user switches from a system keystore to the basic keystore
> > This will never be intentional, right? Why would they change a running system and intentially lessen the security. It could happen, if they somehow reset the env var to default. But then, the warning is useless because Signal will complain anyway.
> 
> Is there a nice way to check if this is the first install? Then do this instead of the warning_shown file.
> 
> > The difference is previously the native package didn't support it and official builds used plaintext as well. Now the official ones don't do anymore
> > They fallback to basic just as we do here. Apparently, the implementation is a bit broken, so it can't find the correct backend correctly. This database error is not exclusive to flatpak.

config.json, among others, are generated on first launch.

`if [[ ! -f "${XDG_RUNTIME_DIR}"/app/${FLATPAK_ID}/config/Signal/config.json ]]` should do it?

**Comment by bbhtt** [comment]
if it's broken on official ones as well you could use the first install idea, just check if the config folder or some file in it exists.

On first install it won't exist unless signal is executed which is at the very last line of the script.

File: signal-desktop.sh [hunk 1]
```diff
@@ -53,11 +53,12 @@ case "${SIGNAL_PASSWORD_STORE}" in
...
 
+# Warn the user about plaintext password
+# - if the user chose basic (this is the default)
+# - and Signal starts for the first time
 if [[ "${SIGNAL_PASSWORD_STORE}" == "basic" ]]; then
-    if [[ -f "${XDG_CACHE_HOME}"/warning-shown ]]; then
-        rm "${XDG_CACHE_HOME}"/warning-shown || true
-    else
-        report_warning
+    if [[ ! -f "${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}/config/Signal/config.json" ]]; then
```
Comment (bbhtt) [inline signal-desktop.sh:60]: This should be $XDG_CONFIG_DIR ? runtime dir is `/run/user/1000`

**Comment by bermeitinger-b** [comment]
This will only warn upon the first run. (However, we differ here from upstream.)

Also, this is generally a workaround, The secrets portal should provide the correct backend name by its run.

File: signal-desktop.sh (hunk 1 above)
Comment (minosimo) [reply signal-desktop.sh:60]: Yes my bad!!

Comment (bbhtt) [reply signal-desktop.sh:60]: "${XDG_CONFIG_DIR}"/Signal/config.json

Comment (bermeitinger-b) [reply signal-desktop.sh]: It's XDG_CONFIG_HOME

**Comment by bbhtt** [comment]
You can drop the else condition that touch es the file in lines 26-28

**Pull request merged**